### PR TITLE
Config for compatiblity to ASDG Joint Rails

### PR DESCRIPTION
**Pull request by OnkelDisMaster**
Added the  scopes and railitems to the according ASDG rails.

### CODE REVIEW

**Comment by jonpas** [comment]
Fix indentation to 4 spaces.

**Comment by jaynus** [comment]
This requires a 3rd party addon; we cannot require third party addons. Is there a way for compat to exist without requiring external classes and addons? 

Otherwise, this needs to be implemented in the other addon; not ours.

**Comment by OnkelDisMaster** [comment]
well thats why its on the optionals... commy told me in #477 to do that

**Comment by OnkelDisMaster** [comment]
indentation to 4 spaces done!

**Comment by rlex** [comment]
Working great for me, but i do not see PiP scopes in list

File: optionals/asdg_comp/config.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    {
+	ACE_optic_Hamr_2D = 1;
+	ACE_optic_Hamr_PIP = 1;
+	ACE_optic_Arco_2D = 1;
+	ACE_optic_Arco_PIP = 1;
+	ACE_optic_MRCO_2D = 1;
+	ACE_optic_SOS_2D = 1;
+	ACE_optic_SOS_PIP = 1;
+	ACE_optic_LRPS_2D = 1;
+	ACE_optic_LRPS_PIP = 1;
```
Comment (commy2) [inline optionals/asdg_comp/config.cpp]: Remove the tabs. 4 spaces intendation only.

File: optionals/asdg_comp/config.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    	ACE_acc_pointer_red = 1;
+    	ACE_acc_pointer_green = 1;
+    };
+};
+
+class asdg_OpticRail;
+class asdg_OpticRail1913: asdg_OpticRail
+{
+    class compatibleItems
+    {
```
Comment (commy2) [inline optionals/asdg_comp/config.cpp]: opening curly bracket goes at the end of the line.
` class Schinken {`

**Comment by commy2** [comment]
I have no issue with putting this into the optional files folder or something.
Thi PR needs formattig though.

**Comment by rlex** [comment]
Can you extend make.py to build optional pbos as well?
Something like
optionals/compatibility/ace_asdg_comp/
That will build on release.
Can be good for RHS/BWmod/etc compatibility PBOs (i remember AGM had AGM_Compatibility, which, however, was never released)

**Comment by commy2** [comment]
I have no idea. I'd just move the optional folder into the same folder a everything else, binarize and then move it back.

**Comment by commy2** [comment]
This PR requires: https://github.com/acemod/ACE3/pull/486/files

Still needs formatting

**Comment by commy2** [comment]
:+1:

**Pull request merged**